daemon/chaintopology.c

```diff
@@ -237,7 +238,7 @@ static void try_broadcast(struct lightningd_state *dstate,
 	this_tx = txs[num_txs-1];
```

Comment:
You're right!  The error reporting is on the wrong tx.  Always was, it seems.  I'll append a patch which fixes this (leave the last tx on, then remove it before going to the next).